STYLE.md

```diff
@@ -0,0 +1,125 @@
+
+**Don't:**
+
+```python
+from subprocess import run
+from ops import CharmBase, PebbleReadyEvent
+import typing
+
+class MyCharm(CharmBase):
+    counts: typing.Optional[typing.Tuple[str, int]]
+
+    def _pebble_ready(self, event: PebbleReadyEvent):
+        run(['echo', 'foo'])
+```
+
+**Do:**
+
+```python
+import subprocess
+import ops
+from typing import Optional, Tuple
+
+class MyCharm(ops.CharmBase):
+    counts: Optional[Tuple[str, int]]
+
+    def _pebble_ready(self, event: ops.PebbleReadyEvent):
+        run(['echo', 'foo'])
+```
+
+
+### Avoid nested comprehensions and generator expressions
+
+"Flat is better than nested."
+
+**Don't:**
+
+```python
+units = [units for app in model.apps for unit in app.units]
+
+for current in (
+    status for status in pebble.ServiceStatus if status != pebble.ServiceStatus.ACTIVE
+):
+    ...
+```
+
+**Do:**
+
+```python
+units = []
+for app in model.apps:
+    for unit in app.units:
+        units.append(unit)
+
+for current in pebble.ServiceStatus:
+    if status == pebble.ServiceStatus.ACTIVE:
+        continue
+    ...
+```
+
+
+### Compare enum values by equality
+
+This is six of one, half a dozen of the other, but we've decided that saying `if color == Color.RED` is nicer than `if color is Color.RED` as `==` is more typical for integer- and string-like values, and if you do use an `IntEnum` or `StrEnum` you should use `==` anyway.
+
+Note that the [Enum HOWTO](https://docs.python.org/3/howto/enum.html#comparisons) first says they should be compared by identity, but then shows that equality is defined too, and so kind of adds to the confusion. See also this [StackOverflow question](https://stackoverflow.com/questions/25858497/should-enum-instances-be-compared-by-identity-or-equality) and [some discussion/disagreement](https://github.com/pylint-dev/pylint/issues/5356).
+
+**Don't:**
+
+```python
+if status is pebble.ServiceStatus.ACTIVE:
+    print('Running')
+
+if status is not pebble.ServiceStatus.ACTIVE:
+    print('Stopped')
+```
+
+**Do:**
+
+```python
+if status == pebble.ServiceStatus.ACTIVE:
+    print('Running')
+
+if status != pebble.ServiceStatus.ACTIVE:
+    print('Stopped')
+```
+
+
+## Docs and docstrings
+
+### Use British English
+
+[Canonical's documentation style](https://docs.ubuntu.com/styleguide/en/) is to use British spellings, and we try to follow that here. For example, "colour" rather than "color", "labelled" rather than "labeled", "serialise" rather than "serialize", and so on.
+
+It's a bit less clear when we're dealing with code and APIs, as those normally use US English, for example, `pytest.mark.parametrize`, and `color: #fff`.
+
+
+### Spell out abbreviations
+
+Prefer spelling out abbreviations and acronyms in docstrings, for example, "for example" rather than "e.g.", "that is" rather than "i.e.", "and so on" rather than "etc", and "unit testing" rather than UT.
```

Comment:
I haven't seen anyone use UT so...

I think there's a balance to be struck, e.g. "Juju RPC" instead of "jRPC" or "Juju remote procedure call" or "Juju (no JSON!) [asynchronous] remote procedure call [protocol over WebSocket]".

Reply:
P.S. Canonical style guide already has a section on this.

How about:

```suggestion
Prefer spelling out abbreviations and acronyms in docstrings, for example, "and so on" rather than "etc." as documented in https://docs.ubuntu.com/styleguide/en/#latin-words-and-phrases
```

Reply:
+1 on linking out, and avoiding the Latin is most important I think.

I do think "Juju RPC" is better than "jRPC" ... but I would use "JSON" and "RPC" still.

Reply:
Yeah, definitely agreed there's a balance to be struck. I've added this sentence that hopefully clarifies that intent:

> However, it's okay to use acronyms that are very well known in our domain, like HTTP or JSON or RPC.